### DISCUSSION TITLE
Improve displaying of error when viewing a content item

### DISCUSSION THREAD

**Discussion by pieterdd** [directus/directus#24917] [Feature Requests]
### Describe the Bug

When I load http://localhost:8055/admin/content/event/1, a GET call to http://localhost:8055/items/event/1 returns an HTTP 500. The UI reacts to the 500 incorrectly:

![image](https://github.com/directus/directus/assets/1407980/fc4403fa-86c2-4ed2-ac5c-d024a7e3bd85)

A correct error would be something along the lines of "A technical error has occurred".

### To Reproduce

Haven't figured out yet why the HTTP 500 is happening or how to reproduce it, but that's beyond the scope of this bug report. As for the frontend incorrectly reacting to the HTTP 500:

- Create a data model
- Create an item within that data model
- Open its edit page and use a mocking tool like HTTP Toolkit to force the XHR call to `http://localhost:8055/items/[data model]/[id]` to respond with HTTP 500

### Directus Version

10.7.1

### Hosting Strategy

Self-Hosted (Docker Image)

**Comment by azrikahar** [comment]
I was able to "induce" said error intentionally using a filter hook on items.read, with the following snippet as an example:

```js
import { ServiceUnavailableError } from '@directus/errors';

export default function ({ filter }, _context) {
	filter('test.items.read', (records, _meta) => {
		if (records.length === 1 && records[0].id === 1) {
			throw new ServiceUnavailableError({
				service: 'test.items.read filter',
				reason: `Reading the first item should throw error for testing purposes`,
			});
		}
	})
};
```

which causes the Page Not Found issue to happen whenever I view the first item (id = 1) in `test` collection.

Upon looking it abit further, this is because `content-not-found` component is displayed using `v-if` when there's `error`.

This logic/design should be updated, probably referencing the existing error UI shown in layouts such as:

https://github.com/directus/directus/blob/d54418bb39c74f8c31fe0386dd0367f923eb457a/app/src/layouts/tabular/tabular.vue#L290-L300

that is specifically for `error` case rather than a "catch-all" like how it is in content item.
